src/README.md

```diff
@@ -0,0 +1,27 @@
+## Welcome to the Babel Codebase
+
+Babel is broken down into three parts:
+
+- Parsing
+- Transformation
+- Generation
+
+**Parsing** is the process of turning a piece of code into an AST (Abstract
+Syntax Tree) which is a tree-like object of nodes that _describes_ the code.
+This makes it easier to transform the code over direct string manpulation.
+
+**Transformation** is the process of taking an AST and manipulating it into
+a new one. For example, Babel might take an ES2015 ArrowFunction and transform
```

Comment:
@thejameskyle That's a great point about them showing up while you browse. That would also make it easy to link to other docs from there. I think an advantage to putting most of the content under [`/doc`](https://github.com/babel/babel/tree/master/doc) is that then there's a centralized location people can look in to see a collection of a bunch of internals documentation. But I won't quibble about where the docs live, I think it's great that you're adding them, and whether the `README`s contain a bunch of links or inline a bunch of the content they seem like a great idea. Another option is to put links to the various `README`s in [`/doc/index.md`](https://github.com/babel/babel/tree/master/doc/index.md)